### ISSUE TITLE
DIGITS Torch not recognizing CuDNN or NCCL

### ISSUE THREAD

**Issue by michaelholm-ce** [NVIDIA/DIGITS#1196]
I'm getting the following warnings when running a torch model training job via DIGITS:
```
2016-10-20 16:04:57 [20161020-160446-8964] [WARNING] Train Torch Model unrecognized output: Using CuDNN backend
2016-10-20 16:04:57 [20161020-160446-8964] [WARNING] Train Torch Model unrecognized output: warning: could not load nccl, falling back to default communication
```
I'm concerned, because I believe both CuDNN and NCCL will greatly speed up large jobs.  

Background:  I have installed and compiled Caffe 0.14.0-rc.3 with NCCL and CUDNN flags on.  I am using DIGITS 5.0.0-rc.1.  I have run ```sudo luarocks install cudnn```, and have installed the appropriate NCCL, following all instructions on ```https://github.com/NVIDIA/nccl```.  I have installed CUDA 7.5.

**Comment by lukeyeager** [comment]
It looks like you are already using cuDNN.

Did you install the Torch bindings for nccl?
> `luarocks install "https://raw.githubusercontent.com/ngimel/nccl.torch/master/nccl-scm-1.rockspec"`
https://github.com/NVIDIA/DIGITS/blob/digits-5.0/docs/BuildTorch.md#extra-dependencies

**Comment by michaelholm-ce** [comment]
Thank you for pointing me to the additional nccl torch bindings -- that solved the loading problem.  

I took ```unrecognized output:  Using CuDNN backend``` to mean that DIGITS did not recognize CUDNN, and that therefore, CUDNN was not being used.  Am I misreading that line?

**Comment by michaelholm-ce** [comment]
Note: I have ```libcudnn.so.5``` in ```/usr/local/cuda/lib64```, which is in my ```LD_LIBRARY_PATH```.

**Comment by lukeyeager** [comment]
> I took unrecognized output: Using CuDNN backend to mean that DIGITS did not recognize CUDNN, and that therefore, CUDNN was not being used. Am I misreading that line?

Yes, you're misreading it. But I don't blame you because it's confusing :confused: 

Torch is printing a little message telling you that it's using cuDNN. Then DIGITS parses the Torch output and looks for messages it recognizes like "loss = 2.0". We haven't added that message as an expected output, so we just print it in the log in case it's important.

**Comment by michaelholm-ce** [comment]
That is very funny!  Thanks for explaining, @lukeyeager.
